### PR TITLE
Fix alternator validation

### LINKED ISSUE #1790: Alternator validation doesn't validate certificate options
#1719  didn't fully wire Alternator certificate validation. We need to fix it before we release v1.12 as we can't restrict validation after it goes live.

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
Fixes dead forgotten validation code from #1719 

Resolves #1790

### CODE REVIEW

File: pkg/api/scylla/validation/cluster_validation_test.go [hunk 1]
```diff
@@ -320,6 +320,120 @@ func TestValidateScyllaCluster(t *testing.T) {
...
+				}
+				return cluster
+			}(),
+			expectedErrorList: field.ErrorList{
+				&field.Error{Type: field.ErrorTypeNotSupported, Field: "spec.alternator.servingCertificate.type", BadValue: v1.TLSCertificateType("foo"), Detail: `supported values: "OperatorManaged", "UserManaged"`},
+			},
+			expectedErrorString: `spec.alternator.servingCertificate.type: Unsupported value: "foo": supported values: "OperatorManaged", "UserManaged"`,
+		},
+		{
+			name: "alternator cluster with invalid certificate type",
```
Comment (zimnx) [inline pkg/api/scylla/validation/cluster_validation_test.go:350]: this one is duplicated

File: pkg/api/scylla/validation/cluster_validation_test.go [hunk 2]
```diff
@@ -320,6 +320,120 @@ func TestValidateScyllaCluster(t *testing.T) {
...
+						},
+					},
+				}
+				return cluster
+			}(),
+			expectedErrorList: field.ErrorList{
+				&field.Error{Type: field.ErrorTypeInvalid, Field: "spec.alternator.servingCertificate.operatorManagedOptions.additionalIPAddresses", BadValue: []string{"0.not-an-ip.0.0"}, Detail: `must be a valid IP address, (e.g. 10.9.8.7 or 2001:db8::ffff)`},
+			},
+			expectedErrorString: `spec.alternator.servingCertificate.operatorManagedOptions.additionalIPAddresses: Invalid value: []string{"0.not-an-ip.0.0"}: must be a valid IP address, (e.g. 10.9.8.7 or 2001:db8::ffff)`,
+		},
```
Comment (zimnx) [inline pkg/api/scylla/validation/cluster_validation_test.go:436]: lets add test validating secretName added in this commit to have full coverage

File: pkg/api/scylla/validation/cluster_validation_test.go (hunk 1 above)
Comment (tnozicka) [reply pkg/api/scylla/validation/cluster_validation_test.go:351]: fixed, thx

File: pkg/api/scylla/validation/cluster_validation_test.go (hunk 2 above)
Comment (tnozicka) [reply pkg/api/scylla/validation/cluster_validation_test.go:437]: added

**Pull request merged**